src/charm.py

```diff
@@ -33,11 +34,16 @@ def __init__(self, *args):
         super().__init__(*args)
 
         self.logger: logging.Logger = logging.getLogger(__name__)
+        self._namespace = self.model.name
+        self._app_name = self.model.app.name
```

Comment:
There are other places where we also use `app.name`? Should we replace this everywhere?